### PR TITLE
Add a console warning when browser bundle is loaded in node.

### PR DESCRIPTION
**Pull request by hsubox76**
Partially addresses issue https://github.com/firebase/firebase-js-sdk/issues/1415.

Open to suggestions on what the message should recommend to the user.

Tried a rollup node test project and browser webpack test project.  Warning is seen in node, not seen in browser.

### CODE REVIEW

File: packages/app/index.ts [hunk 1]
```diff
@@ -18,6 +18,23 @@
...
+let isNode = false;
+try {
+  isNode =
+    Object.prototype.toString.call(global.process) === '[object process]';
+} catch (e) {}
+
+isNode &&
+  console.warn(`
+Warning: This is a browser-targeted Firebase bundle but it appears it is being run in a Node
+environment.  This may be because you are using ES module resolution and the Node bundle is
```
Comment (Feiyang1) [inline packages/app/index.ts]: It's not clear to me what ES module resolution is. It's probably better to just state please use the "main" field in package.json.
We can also add the link to webpack doc - https://webpack.js.org/configuration/resolve/#resolvemainfields
and rollup doc - https://github.com/rollup/rollup-plugin-node-resolve

Comment (hsubox76) [reply packages/app/index.ts]: Modified the text.

**Comment by arpowers** [comment]
The solution you guys are proposing is not going to work. Also you give no ways of debugging this issue since Webpack is compiling normally with the 'node' target set.

**Comment by arpowers** [comment]
is says clearly here: 
https://webpack.js.org/configuration/resolve/#resolvemainfields

where you link that a node targeted environment will load 'module' first, YET, you've added this warning to the index.esm.js entry (you also list module: index.esm.js)

**Comment by Feiyang1** [comment]
Thank you for the feedback. And sorry you had a bad experience.

To clarify how fields are used in firebase currently:
main: cjs build for Node
module: esm build for browser
browser: cjs build for browser
react-native: cjs build for react-native

Each firebase SDK (except storage) does something different for Node, so the browser version should really not be used for Node environment. and it could cause issues down the road if you are already using it, so I think the warning message is warranted.

We have both cjs and esm build for browser is because we'd like to support as many bundlers as possible, for example, browserify only works with cjs natively.

I'd like to have a `browser-module` field, so I can use `module` field to point to node version, but I don't think there is something like this at the moment, so we chose to use the module field for the browser build.

I'm open to suggestions, but given the constraint of a single module field, I don't know what more we can do than giving a warning message when a bundle is used in the unintended environment.

I'm not convinced that changing the global `resolve.mainFields` to read `main` field first is going to cause issue for other packages as all node packages should have a cjs build pointed by the `main` field.

If you are really worried about it, you could do something like: 
```javascript
  resolve: {
    alias: {
      firebase$: path.resolve(__dirname, 'node_modules/firebase/dist/index.node.cjs.js')
    }
  }
```

Anyway, thanks for voicing your concern. If you'd like to continue the discussion, please open an issue, so we can track it.

**Pull request merged**